packages/frametricks/init.lua

```diff
@@ -153,12 +153,28 @@ function package:registerCommands ()
 
   self:registerCommand("showframe", function (options, _)
     local id = options.id or SILE.typesetter.frame.id
-    if id == "all" then
-      for _, frame in pairs(SILE.frames) do
-        SILE.outputter:debugFrame(frame)
+    options.allpages = options.allpages or true
+
+    local show = function()
+      if id == "all" then
```

Comment:
Even local functions should not use variables outside of their scope with a documented reason for needing to. This is called a side effect and has lots of downsides, not to mention will make porting harder down the road. Just add an id field to the function and pass it through explicitly.